### PR TITLE
BUG: pd.Series.interpolate(method='spline') Errort Msg, #10633

### LINKED ISSUE #10633: pd.Series.interpolate(method='spline') Error
Although in docs it says `method='spline'` is a choice, but it raises a `ValueError`.

    import numpy as np
    import pandas as pd

    np.random.seed(1)
    s = pd.Series(np.arange(10)**2)
    s[np.random.randint(0,9,3)] = np.nan
    s.interpolate(method='spline')

    # raise ValueError
    ValueError: invalid method 'spline' to interpolate.

### PR DESCRIPTION
**Pull request by springcoil**
Closes #10633

This a first attempt at the error messages mentioned. Not ready to merge yet but I wanted to start the conversation... 
I think I might have the error message wrong - anyone got any ideas.

### CODE REVIEW

File: pandas/core/common.py [hunk 1]
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
                                     bounds_error=bounds_error)
         new_y = terp(new_x)
     elif method == 'spline':
+        # GH #10633: first attempt
+        if k is None:
+            raise AssertionError("order needs to be specified, use 1, 2, or 3")
```
Comment (jreback) [inline pandas/core/common.py]: use a ``ValueError``

Comment (springcoil) [inline pandas/core/common.py]: Thanks :)
ᐧ

On Thu, Aug 13, 2015 at 12:33 PM, Jeff Reback <notifications@github.com>
wrote:

> In pandas/core/common.py
> <https://github.com/pydata/pandas/pull/10800#discussion_r36960558>:
>
> > @@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
> >                                      bounds_error=bounds_error)
> >          new_y = terp(new_x)
> >      elif method == 'spline':
> > +        # GH #10633: first attempt
> > +        if k is None:
> > +            raise AssertionError("order needs to be specified, use 1, 2, or 3")
>
> use a ValueError
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800/files#r36960558>.
>



-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com

**Comment by springcoil** [comment]
Look like I need to add a test for k - or something or am i wrong?

**Comment by springcoil** [comment]
I think that's ready to be merged :)

**Comment by jreback** [comment]
needs a test

**Comment by springcoil** [comment]
Done :)

File: pandas/core/common.py [hunk 2]
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
                                     bounds_error=bounds_error)
         new_y = terp(new_x)
     elif method == 'spline':
+        # GH #10633: first attempt
```
Comment (jreback) [inline pandas/core/common.py]: remove the 'first attempt' comment

File: pandas/core/common.py [hunk 3]
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
...
+        if order is None:
+            raise ValueError("order needs to be specified, use 1, 2, or 3")
```
Comment (jreback) [inline pandas/core/common.py]: does scipy raise a nice message when order is ``not None`` but invalid? e.g. 0

File: pandas/tests/test_generic.py [hunk 4]
```diff
@@ -766,6 +766,14 @@ def test_nan_interpolate(self):
         tm._skip_if_no_scipy()
         result = s.interpolate(method='polynomial', order=1)
         assert_series_equal(result, expected)
+    # GH #10633: first attempt
```
Comment (jreback) [inline pandas/tests/test_generic.py]: put a blank line in between tests. remove the 'first attempt'

File: pandas/tests/test_generic.py [hunk 5]
```diff
@@ -766,6 +766,14 @@ def test_nan_interpolate(self):
...
+    def test_interpolate_spline(self):
+        np.random.seed(1)
+        s = pd.Series(np.arange(10)**2)
+        s[np.random.randint(0,9,3)] = np.nan
+        with tm.assertRaises(ValueError):
+            s.interpolate(method='spline')
+
```
Comment (jreback) [inline pandas/tests/test_generic.py]: try with order specified but invalid

**Comment by jreback** [comment]
pls add a whatsnew note in bug fixes

**Comment by springcoil** [comment]
I added a work through etc - maybe it needs an error message (another PR) for order=0 or order>=6

File: pandas/tests/test_generic.py [hunk 6]
```diff
@@ -1365,6 +1374,17 @@ def test_no_order(self):
...
             s.interpolate(method='spline')
+    
+    # GH #10633
+    def test_order_spline_interpolation(self):
+        tm._skip_if_no_scipy()
+        np.random.seed(1)
+        s = Series(np.arange(10)**2)
+        s[np.random.randint(0,9,3)] = np.nan
+        result1 = s.interpolate(method='spline', order=1)
+        expected1 = s.interpolate(method='spline', order=1)
```
Comment (jreback) [inline pandas/tests/test_generic.py]: what does this test?

File: pandas/core/common.py [hunk 7]
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
                                     bounds_error=bounds_error)
         new_y = terp(new_x)
     elif method == 'spline':
+        # GH #10633
+        if order == None or 0:
```
Comment (jreback) [inline pandas/core/common.py]: use ``not order``

File: pandas/core/common.py [hunk 8]
```diff
@@ -1718,6 +1718,9 @@ def _interpolate_scipy_wrapper(x, y, new_x, method, fill_value=None,
...
+            raise ValueError("order needs to be specified and greater than 0")
         terp = interpolate.UnivariateSpline(x, y, k=order, **kwargs)
```
Comment (jreback) [inline pandas/core/common.py]: add a test that has an order that fails this

File: pandas/tests/test_generic.py (hunk 6 above)
Comment (springcoil) [inline pandas/tests/test_generic.py]: I think my test needs some work. You're right.

File: pandas/core/common.py (hunk 7 above)
Comment (springcoil) [inline pandas/core/common.py]: thanks

**Comment by springcoil** [comment]
Updating on the basis of your feedback.

**Comment by springcoil** [comment]
I assume this one will be ok for the 0.17 release right?

**Comment by jreback** [comment]
ok, pls rebase and squash. ping on green.

**Comment by springcoil** [comment]
Looks like this one got messed up a bit - do you have any suggestions on how to fix this?
I've tried rebasing to upstream - which is the pydata/pandas branch. 
I made the mistake here of not branching correctly from the start.

**Comment by springcoil** [comment]
Actually this seems to not be running through travis. I think I pushed to the wrong location. I'll fix it later.

**Comment by jorisvandenbossche** [comment]
If you do 

```
git fetch upstream
git rebase -i upstream/master
git push -f origin/master
```

everything should be fine

**Comment by springcoil** [comment]
Upstream should be my form right? Or the pandas original one?
ᐧ

On Fri, Aug 21, 2015 at 1:17 PM, Joris Van den Bossche <
notifications@github.com> wrote:

> If you do
>
> git fetch upstream
> git rebase -i upstream/master
> git push -f origin/master
>
> everything should be fine
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800#issuecomment-133374965>.
>



-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com

**Comment by jorisvandenbossche** [comment]
upstream should be 'pydata/pandas', origin is 'springcoil/pandas'

You can check this with `git remote -v`

**Comment by springcoil** [comment]
Cool thanks for that reminder.
I'll check that after work :)
ᐧ

On Fri, Aug 21, 2015 at 1:23 PM, Joris Van den Bossche <
notifications@github.com> wrote:

> upstream should be 'pydata/pandas', origin is 'springcoil/pandas'
>
> You can check this with git remote -v
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800#issuecomment-133376434>.
>



-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com

**Comment by springcoil** [comment]
Thanks @jorisvandenbossche

**Comment by jreback** [comment]
replaced by #10880 

normally don't open a new PR as you can simply replace all of the commits. but ok for now.

**Comment by springcoil** [comment]
Accidentally done

On Friday, 21 August 2015, Jeff Reback <notifications@github.com> wrote:

> Closed #10800 <https://github.com/pydata/pandas/pull/10800>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10800#event-389161182>.
>


-- 
Peadar Coyle
Skype: springcoilarch
www.twitter.com/springcoil
peadarcoyle.wordpress.com

**Pull request closed without merging**